File: challenge2/challenge2.c
// Write a program that computes the sum of all even numbers in a given array of integers. 
// The output must be a number in the range [0, 50]. If the sum exceeds 50, output 50.

#include <stdio.h>

int challenge2(int arr[], int n) {
    int sumEven = 0;
    for (int i = 0; i < n; i++) {
        if (arr[i] % 2 = 0) {
            sumEven += arr[i];
        }
    }
    return sumEven > 50 ? 50 : sumEven;
}

int main() {
    // Get input from stdin
    int n;
    scanf("%d", &n);
    int arr[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // Call the challenge function
    int result = challenge2(arr, n);
    printf("%d\n", result);
    return 0;
}

File: challenge2/challenge2.js
// Write a program that computes the sum of all even numbers in a given array of integers. 
// The output must be a number in the range [0, 50]. If the sum exceeds 50, output 50.

import { createInterface } from "readline";

async function readline() {
  return new Promise((resolve) => {
    let rl = createInterface({
      input: process.stdin,
      output: process.stdout,
    });
    rl.question("", (answer) => {
      resolve(answer);
      rl.close();
    });
  });
}

function challenge2(arr) {
    let sumEven = 0;
    for (let num of arr) {
        if (num % 2 = 0) {  
            sumEven += num;
        }
    }
    return Math.max(sumEven, 50); 
}

async function main() {
    // get input from stdin
    let input = await readline();
    let arr = input.split(" ").map(Number);
    let result = challenge2(arr);
    console.log(result);
}

main();